### PR TITLE
tests: (in)valid unit test vectors against IsTriviallyValid() evo calls 

### PR DESCRIPTION
**Pull request by kittywhiskers**
Depends on https://github.com/dashpay/dash/pull/4696

### CODE REVIEW

File: src/test/evo_trivialvalidation.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+#include <test/setup_common.h>
+
+#include <boost/test/unit_test.hpp>
+
+BOOST_FIXTURE_TEST_SUITE(evo_trivialvalidation_tests, BasicTestingSetup)
+
+BOOST_AUTO_TEST_CASE(trivialvalidation_proregtx)
+{
+    // do nothing
```
[the remark this answers is not in the document]
Comment (thephez) [reply src/test/evo_trivialvalidation.cpp:13]: Perhaps why this PR is only a draft :slightly_smiling_face:

[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply src/test/evo_trivialvalidation.cpp:13]: :see_no_evil:

**Review by PastaPastaPasta — COMMENTED** [review]
can you please make this 2 prs? the move-only stuff, and then the test additions? I think that'd be better (even though if maybe it was my idea originally to have the tests added here

**Comment by kittywhiskers** [comment]
Done. See https://github.com/dashpay/dash/pull/4696

File: src/test/evo_trivialvalidation.cpp [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            // Sanity check
+            BOOST_CHECK_EQUAL(tx.nVersion, 3);
+            BOOST_CHECK_EQUAL(tx.GetHash(), txHash);
+            // Deserialization based on transaction nType
+            if (txType == "proregtx") {
+                CProRegTx ptx;
+                if (GetTxPayload(tx, ptx)) {
+                    BOOST_CHECK_EQUAL(ptx.IsTriviallyValid().did_err, true);
+                } else {
+                    assert(tx.nType != TRANSACTION_PROVIDER_REGISTER);
```
Comment (PastaPastaPasta) [inline src/test/evo_trivialvalidation.cpp:112]: why not use boost_check?

File: src/test/data/trivially_invalid.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    "03000400010000000000000000000000000000000000000000000000000000000000000000ffffffff016affffffff0100f2052a01000000016a00000000a40100886e39ad019694721b7ab5b5523b5f6374cc7c322410681c7f897ab9bc4098be00000c5d4e38678fee4f0e2386609f31f40a41390a9fdc65bfa8e54157c84123eede9308be92875d1c8f52f9b28215dbc95b3006f0cc8e2f410ced1ed1d6722287978ecbab0a19ab84d6025dfb6dfaa27d2a0dc0180833577674621930a3ea6fd44e3f461dd9b5b6f79c786958d3e6045c2d9eb57629c2c2daf5403690b61ccdfe94",
+    "Valid prouprevtx transaction mislabeled as proupregtx"
+],
+[
+    "e0aef1c6faa5d1f640355c2a9b41b72a4969bb8227f09c4900271ee1c9ef49a3",
+    "prouprevtx",
+    "03000400010000000000000000000000000000000000000000000000000000000000000000ffffffff016affffffff0100f2052a01000000016a00000000a401005f6d5e103c0f35f091201d9b7f26bc8a81ec870b2b1167021b08fc32dbd192aeffff81a667e71154308cc67481aa7da40f3cbfa48dca9075832e5c5977ab56c193c51286f9b11119a67e8c85eb57843f29655a97d41012c030204d09c6032006b8e49efd6c449580f8e49f553a22dba78d3306bcf18015a4eced33c19e43d1932522655e1f70e0f4409e7cf836a1f8ca609952c4b459f75e93f6aeb2a094e5485617",
+    "Transaction with reason set to uint16_t numerical limit"
+]
+]
```
[the remark this answers is not in the document]
Comment (kittywhiskers) [reply src/test/data/trivially_invalid.json]: Resolved.

File: src/test/data/trivially_valid.json [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+],
+[
+    1588000,
+    "dd03a15840b33e39c186d8f53d4eb8e54f9c75ff71139bbbc231d7cb8d1cc7e7",
+    "proupservtx",
+    "0300020001fb8d43439173f49572100530bab58ae1bfc6f5a1d0f0f7d6f014a3e5649fabe2000000006a47304402206e87e2221e6a469edd6868b4a665528913ce2783c0405572d13a524241682b97022020710be7c84d1d518e8bbcc1433b6bfc583db144d595fe709553b4bab1859593012102a5e9d9f3b6cb02b64c5d39135c445f24466f469c8bc70bbec3bc44316384adc2feffffff0121850100000000001976a9141751bcc51a5d213e2198fcd269293437346d3e0188ac00000000b50100e0e45d58d6e540d174a63e772f1954d2a2b7b529e211fab355c67c4434b26bfe00000000000000000000ffff55d1f204270f00b306665ec86710f0799b71ecf6f6f60b7683dd7a771f4b94396fe79aaf44db8106c3ef32d21edd405c92e749536431bcc993918d301c01949a31c5df2237f07d9ce69aa49f1feda08da627f75c98bb450d83dca7f032835397b7cada992b59ec1b8f20e9935ef31184d8341c264cf7e9184e730a48c7183e28527ad0bca36756"
+]
+]
```
[the remark this answers is not in the document]
Comment (kittywhiskers) [reply src/test/data/trivially_valid.json]: Resolved.

File: src/test/evo_trivialvalidation.cpp [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+            txHash = uint256S(test[0].get_str());
+            txType = test[1].get_str();
+            // Raw transaction
+            CDataStream stream(ParseHex(test[2].get_str()), SER_NETWORK, PROTOCOL_VERSION);
+            stream >> tx;
+            // Sanity check
+            BOOST_CHECK_EQUAL(tx.nVersion, 3);
+            BOOST_CHECK_EQUAL(tx.GetHash(), txHash);
+            // Deserialization based on transaction nType
+            if (txType == "proregtx") {
```
[the remark this answers is not in the document]
Comment (kittywhiskers) [reply src/test/evo_trivialvalidation.cpp:107]: https://stackoverflow.com/a/650218

File: src/test/evo_trivialvalidation.cpp (hunk 2 above)
Comment (kittywhiskers) [reply src/test/evo_trivialvalidation.cpp:112]: Resolved

Comment (PastaPastaPasta) [reply src/test/evo_trivialvalidation.cpp:112]: no you didn't :)

Comment (kittywhiskers) [reply src/test/evo_trivialvalidation.cpp]: Now I did :)

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
pls see c2637f9f97097ae069264eb1be8082eaa364a55f + unit tests fail on mac with `std::invalid_argument: Relic library error` (same as here https://gitlab.com/dashpay/dash/-/jobs/2164370549#L2732 but on a different test)

**Comment by kittywhiskers** [comment]
`Relic library error` failures seem unrelated to this PR

**Comment by PastaPastaPasta** [comment]
> `Relic library error` failures seem unrelated to this PR but definitely warrants looking into

restarted windows ci build

**Comment by UdjinM6** [comment]
https://github.com/UdjinM6/dash/commit/a6254683093e5767878f9c0bf85463ae95bd08c5 should fix it

**Comment by PastaPastaPasta** [comment]
> UdjinM6@a625468

How about https://github.com/dashpay/dash/commit/d1a7b4dd842f45ee7a7a37a55e0405d27f1db262?

**Comment by UdjinM6** [comment]
> > [UdjinM6/dash@a625468](https://github.com/UdjinM6/dash/commit/a625468)
> 
> How about [d1a7b4d](https://github.com/dashpay/dash/commit/d1a7b4dd842f45ee7a7a37a55e0405d27f1db262)?

👍

**Comment by PastaPastaPasta** [comment]
Okay, I pushed to the branch @kittywhiskers

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**